desc/objectives/_geometry.py

```diff
@@ -459,6 +459,290 @@ def compute(self, params, constants=None):
+        Lower and upper bounds on the objective. Overrides target.
+        Both bounds must be broadcastable to to Objective.dim_f
+    weight : {float, ndarray}, optional
+        Weighting to apply to the Objective, relative to other Objectives.
+        Must be broadcastable to to Objective.dim_f
+    normalize : bool, optional
+        Whether to compute the error in physical units or non-dimensionalize.
+    normalize_target : bool, optional
+        Whether target and bounds should be normalized before comparing to computed
+        values. If `normalize` is `True` and the target is in physical units,
+        this should also be set to True.
+        be set to True.
+    loss_function : {None, 'mean', 'min', 'max'}, optional
+        Loss function to apply to the objective values once computed. This loss function
+        is called on the raw compute value, before any shifting, scaling, or
+        normalization. Note: Has no effect for this objective.
+    deriv_mode : {"auto", "fwd", "rev"}
+        Specify how to compute jacobian matrix, either forward mode or reverse mode AD.
+        "auto" selects forward or reverse mode based on the size of the input and output
+        of the objective. Has no effect on self.grad or self.hess which always use
+        reverse mode and forward over reverse mode respectively.
+    grid : Grid, optional
+        Collocation grid containing the nodes to evaluate at.
+    name : str, optional
+        Name of the objective function.
+
+    """
+
+    def __init__(
+        self,
+        coil,
+        data_keys,
+        target=None,
+        bounds=None,
+        weight=1,
+        normalize=True,
+        normalize_target=True,
+        loss_function=None,
+        deriv_mode="auto",
+        grid=None,
+        name=None,
+    ):
+        self._grid = grid
+        self._data_keys = data_keys
+        super().__init__(
+            things=[coil],
+            target=target,
+            bounds=bounds,
+            weight=weight,
+            normalize=normalize,
+            normalize_target=normalize_target,
+            loss_function=loss_function,
+            deriv_mode=deriv_mode,
+            name=name,
+        )
+
+    def build(self, use_jit=True, verbose=1):
+        """Build constant arrays.
+
+        Parameters
+        ----------
+        use_jit : bool, optional
+            Whether to just-in-time compile the objective and derivatives.
+        verbose : int, optional
+            Level of output.
+
+        """
+        # local import to avoid circular import
+        from desc.coils import CoilSet, MixedCoilSet
+
+        is_mixed_coils = False
+        if isinstance(self.things[0], CoilSet):
+            if isinstance(self.things[0], MixedCoilSet):
+                is_mixed_coils = True
+                coil = self.things[0].coils
+                self._dim_f = len(coil)
+            else:
+                coil = self.things[0].coils[0]
+                self._dim_f = 1
```

Comment:
for possibly nested coilsets you can find the total number of coils with the `tree_flatten` util in JAX (imported in desc.backend):
```python
tree_flatten(things, isleaf = lambda x: isinstance(x, Coil) and not isinstance(x, CoilSet))
```

Reply:
If its a regular `CoilSet` that should work fine because you can re-use transforms for each coil. If it's a mixed coilset you'll have to get the transforms for each one by one

Reply:
Okay so then in compute(), should data for all coils in the regular `CoilSet` be computed or should it just be computed for one coil? I guess all of them should be included since each coil would be optimized independently right?